### PR TITLE
build: Consolidate generator testing projects

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by jeromelaban — COMMENTED** [review]
LGTM, though there's a build failure

File: src/SourceGenerators/Uno.UI.SourceGenerators.Tests/MetadataUpdates/HotReloadWorkspaceProvider.cs [hunk 1]
```diff
@@ -222,9 +224,9 @@ public async Task Initialize(CancellationToken ct)
 					// Build the analyzer document additional data information
 					var analyzerDocumentId = DocumentId.CreateNewId(project.Id);
 
+					// For now, there is no need to customize these for each test.
 					var globalConfigBuilder = new StringBuilder($"""
 						is_global = true
-						# For now, there is no need to customize these for each test.
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators.Tests/MetadataUpdates/HotReloadWorkspaceProvider.cs:227]: This comment is moved up due to https://github.com/dotnet/roslyn/issues/69776

**Comment by Youssef1313** [comment]
@jeromelaban This is green now.

**Comment by jeromelaban** [comment]
@Youssef1313 could you rebase? Thanks!

**Comment by Youssef1313** [comment]
@jeromelaban Done.

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**